vagrant/src/main/java/org/jclouds/vagrant/internal/ImageSupplier.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.vagrant.internal;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Collection;
+
+import org.jclouds.compute.domain.Image;
+import org.jclouds.vagrant.api.VagrantBoxApiFacade;
+
+import com.google.common.base.Function;
+import com.google.common.base.Supplier;
+import com.google.common.collect.Collections2;
+import com.google.inject.Inject;
+
+public class ImageSupplier<B> implements Supplier<Collection<Image>>, Function<String, Image> {
+   private final Function<Collection<B>, Collection<B>> outdatedBoxesFilter;
+   private final VagrantBoxApiFacade.Factory<B> cliFactory;
+   private final Function<B, Image> boxToImage;
+
+   @Inject
+   ImageSupplier(Function<Collection<B>, Collection<B>> outdatedBoxesFilter,
+         VagrantBoxApiFacade.Factory<B> cliFactory,
+         Function<B, Image> boxToImage) {
+      this.outdatedBoxesFilter = checkNotNull(outdatedBoxesFilter, "outdatedBoxesFilter");
+      this.cliFactory = checkNotNull(cliFactory, "cliFactory");
+      this.boxToImage = checkNotNull(boxToImage, "boxToImage");
```

Comment:
Yes, Guice already fails if a value is missing so we'd better remove those null checks. I know there are many constructors like this in the codebase but were trying to avoid the redundant checks in all new contributions.